File: csharp/byps/BProtocolS.cs
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace byps
{
    public class BProtocolS : BProtocol
    {
	    public BProtocolS(BApiDescriptor apiDesc, int negotiatedBypsVersion, long negotiatedVersion, ByteOrder negotiatedByteOrder)
            : base(apiDesc, negotiatedBypsVersion, negotiatedVersion, negotiatedByteOrder)
        {
	    }

	    public BProtocolS(BApiDescriptor apiDesc) 
            : base(apiDesc)
        {
	    }

	    public override BOutput getOutput(BTransport transport) 
        {
            if (negotiatedByteOrder == ByteOrder.UNDEFINED) throw new BException(BExceptionC.INTERNAL, "Protocol object can only be used for input.");
            return new BOutputS(transport, negotiatedBypsVersion, negotiatedVersion, negotiatedByteOrder);
	    }

        public override BOutput getResponse(BTransport transport, BMessageHeader requestHeader)
        {
            if (negotiatedByteOrder == ByteOrder.UNDEFINED) throw new BException(BExceptionC.INTERNAL, "Protocol object can only be used for input.");
		    return new BOutputS(transport, requestHeader);
	    }

        public override BInput getInput(BTransport transport, BMessageHeader responseHeader, ByteBuffer buf)
        {
		    return new BInputS(transport, responseHeader, buf);
	    }

        public override BRegistry getRegistry()
        {
            return apiDesc.getRegistry(BBinaryModel.MEDIUM);
        }
    }
}

File: java/byps/src/byps/BProtocolS.java
package byps;

/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */

import java.nio.ByteBuffer;
import java.nio.ByteOrder;

import byps.BException;

public class BProtocolS extends BProtocol {
	
	public static final BBinaryModel BINARY_MODEL = BBinaryModel.MEDIUM;

	public BProtocolS(BApiDescriptor apiDesc, int negotiatedBypsVersion, long negotiatedVersion, ByteOrder negotiatedByteOrder) {
		super(apiDesc, negotiatedBypsVersion, negotiatedVersion, negotiatedByteOrder);
	}

	public BProtocolS(BApiDescriptor apiDesc) {
		super(apiDesc);
	}

	public BOutput getOutput(BTransport transport, BMessageHeader responseHeader) throws BException {
		if (negotiatedByteOrder == null) throw new BException(BExceptionC.INTERNAL, "Protocol object can only be used for input.");
		if (responseHeader != null) {
			return new BOutputS(transport, responseHeader);
		}
		else {
			return new BOutputS(transport, negotiatedBypsVersion, negotiatedVersion, negotiatedByteOrder);
		}
	}
	
	public BInput getInput(BTransport transport, BMessageHeader header, ByteBuffer buf) throws BException {
		return new BInputS(transport, header, buf);
	}

  public BRegistry getRegistry() {
    return apiDesc.getRegistry(BINARY_MODEL);
  }

}
